src/mon/NVMeofGwTypes.h

```diff
@@ -157,35 +157,58 @@ struct NqnState {
     : nqn(_nqn), ana_state(_ana_state)  {}
   NqnState(
     const std::string& _nqn, const SmState& sm_state,
-    const NvmeGwMonState & gw_created)
-    : nqn(_nqn)  {
+    const NvmeGwMonState & gw_created,  bool encode_gmap_last)
```

Comment:
No.  If you need to modify the structures on encode, make a copy and do it in encode.